publisher-sdk/src/main/java/com/criteo/publisher/logging/Logger.java

```diff
@@ -41,39 +39,45 @@ public Logger(
   }
 
   public void debug(String message, Throwable thrown) {
-    log(Log.DEBUG, message, EMPTY, thrown);
+    log(simpleLogMessage(Log.DEBUG, message, thrown));
   }
 
   public void debug(String message, Object... args) {
-    log(Log.DEBUG, message, args, null);
+    log(formattedLogMessage(Log.DEBUG, message, args));
   }
 
   public void info(String message, Throwable thrown) {
-    log(Log.INFO, message, EMPTY, thrown);
+    log(simpleLogMessage(Log.INFO, message, thrown));
   }
 
   public void info(String message, Object... args) {
-    log(Log.INFO, message, args, null);
+    log(formattedLogMessage(Log.INFO, message, args));
+  }
+
+  public void warning(String message, Object... args) {
+    log(formattedLogMessage(Log.WARN, message, args));
   }
 
   public void error(Throwable thrown) {
-    log(Log.ERROR, null, EMPTY, thrown);
+    log(simpleLogMessage(Log.ERROR, null, thrown));
   }
 
   public void error(String message, Throwable thrown) {
-    log(Log.ERROR, message, EMPTY, thrown);
+    log(simpleLogMessage(Log.ERROR, message, thrown));
   }
 
-  private void log(int level, @Nullable String message, Object[] args, @Nullable Throwable thrown) {
+  public void log(@NonNull LogMessage logMessage) {
```

Comment:
Are you planning on making all the log calls use this method (with the `LogMessage` signature)?

Reply:
I was not planning this but it might be a good idea as logs would be kind of part of the public API. So when adding a new log, it would make us think twice.

I guess we can keep the `Logger#debug` for debug.